src/cli/commands/commands_factory.py

```diff
@@ -0,0 +1,47 @@
+import functools
+from typing import Union
+
+from prompt_toolkit.completion import merge_completers
+
+from tests.global_variables import DefaultVariables
+
+from .. import cli_utils
+from ..completers import DynamicNestedCompleter
+from .command import Command, DummyCommand
+from .env_command import EnvCommand
+from .help_command import HelpCommand
+from .test_command import TestCommand
+
+
+class InvalidCommandError(Exception):
+    pass
+
+
+class CommandFactory:
+    _supported_commands = {
+        "": DummyCommand,
+        "test": TestCommand,
+        "list": EnvCommand,
+        "clear": EnvCommand,
+        "help": HelpCommand,
+    }
+
+    @classmethod
+    def get_command(cls, text: str) -> Union[Command, None]:
+        text = text if text else ""
+        factory = cls._supported_commands.get(text.split(" ")[0])
+        try:
+            return factory(text)
```

Comment:
where this ``factory`` function is defined?

Reply:
On line 32: `factory = cls._supported_commands.get(text.split(" ")[0])`
It takes it from `CommandFactory._supported_commands`